esp/src/eclwatch/ECLSourceWidget.js

```diff
@@ -81,17 +78,29 @@ define([
             if (params.readOnly !== undefined)
                 this.readOnly = params.readOnly;
 
-            this.editor = CodeMirror.fromTextArea(document.getElementById(this.id + "EclCode"), {
-                tabMode: "indent",
-                matchBrackets: true,
-                lineNumbers: true,
-                mode: mode,
-                readOnly: this.readOnly,
-                foldGutter: mode === "xml" ? true : false,
-                gutters: mode === "xml" ? ["CodeMirror-linenumbers", "CodeMirror-foldgutter"] : ["CodeMirror-linenumbers"]
+            var eclContent = dom.byId(this.id + "EclContent");
+            var isVisible = function (domNode) {
+                var bRect = domNode.getBoundingClientRect();
+                return (
+                    bRect.top > 0 && bRect.left > 0 &&
+                    bRect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+                    bRect.left <= (window.innerWidth || document.documentElement.clientWidth)
+                );
+            };
```

Comment:
isVisible could be added to src/Utility.ts?

In a diff project I was using:  
```
        // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent
        // Note:  Will return false for visible===hidden (which is ok as it still takes up space on the page)
       domNode.offsetParent === null
```
Not sure if that is quite the same thing?